ee/clickhouse/views/events.py

```diff
@@ -109,10 +112,17 @@ def list(self, request: Request, *args: Any, **kwargs: Any) -> Response:
         return Response({"next": next_url, "results": result})
 
     def retrieve(self, request: Request, pk: Optional[int] = None, *args: Any, **kwargs: Any) -> Response:
-        query_result = sync_execute(SELECT_ONE_EVENT_SQL, {"team_id": self.team.pk, "event_id": pk},)
-        result = ClickhouseEventSerializer(query_result[0], many=False).data
-
-        return Response(result)
+        try:
```

Comment:
This seems fundamentally like the wrong approach to solve these 500s:

1. Try-catch is an antipattern when it's not needed and will swallow real bugs
2. The code feels very special-casey for what it's doing - hard to expand the rest of the API to have similar behavior

Do we need all these different error details?

```python
query_result = sync_execute(SELECT_ONE_EVENT_SQL, {"team_id": self.team.pk, "event_id": pk},)
if len(query_result) == 0:
  raise NotFound(detail="Event not found.")
result = ClickhouseEventSerializer(query_result[0], many=False).data
return Response(result)
```

Reply:
Well actually the problem I started on with this PR is that `sync_execute` throws an error.

Which is why I went the try-except route. I guess I could create a UUID validator before it though